### ISSUE TITLE
json.encode returns a panic for invalid value types

### ISSUE THREAD

**Issue by Asalem1** [influxdata/flux#3094]
Running the following query returns a `panic` to the UI on OSS (v2):

<img width="1374" alt="Screen Shot 2020-08-10 at 1 24 23 PM" src="https://user-images.githubusercontent.com/19984220/89827738-dbaf2300-db0c-11ea-87d9-8260cb55acf4.png">

```
import "influxdata/influxdb/v1"
import "json"

query = (bucket) => from(bucket: bucket)
  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
  |> first()
  |> v1.fieldsAsCols()
  |> group()
  |> count(column: "result")
  |> map(fn: (r) => ({
      r with _value: json.encode(v: r.result)
  }))

query(bucket: "<bucket-name>") 
```

### Steps to Reproduce

1. Run that query in tools against a bucket with data

### Suspected problem

The panic arose when I tried to `map` and `json.encode` the result

**Comment by rickspencer3** [comment]
I believe that the sample in the docs is wrong as well: 

https://v2.docs.influxdata.com/v2.0/reference/flux/stdlib/json/encode/ ...
```
import "json"

from(bucket: "example-bucket")
  |> range(start: -1h)
  |> map(fn: (r) => ({
      r with _value: json.encode(v: r._value)
  }))
```
I believe that will not work because json.encode returns bytes (as mentioned in the documentation), so must be converted to a string to work:
```
 r with _value: string(v:json.encode(v: r._value))
```

This should raise a sensible type error that prompts the user to cast to a string rather than panic, though.

Finally, it's not clear what is expected to happen, because json encoding for a single number will not create an object, but rather just that number. The json for the number 1, is just 1, for example.

**Comment by jsternberg** [comment]
This was fixed in #3137.
